Terminal.Gui/Application.cs

```diff
@@ -143,6 +143,9 @@ internal static void ResetState ()
 
         // Mouse
         _mouseEnteredView = null;
+        _lastViewButtonPressed = null;
+        _canProcessClickedEvent = true;
+        _isMouseDown = false;
```

Comment:
We already have `MouseFlags.Button1Pressed`. The ENTIRE point of this Issue (#3312) is that we have confused the meaning of `Pressed`. 

Instead of a design that reflects how users actually use the mouse, we have a mess. 

"Pressed" is a past-tense of the verb "Press". The "-ed" indicates the "Press" happened in the past. Thus "Pressed" LITERALLY means "Is Currently Down".

"Released" is also past-tense. It means "Is No Longer Down".

I believe the terminology we should use (and the designers of Windows, WinForms, Mac Carbon, and most other UI libraries I've worked with) should distinguish, clearly, between 

- The button WAS not pressed, but it just became pressed. Down makes the most sense to me, but the verb Press works too. 
- The button was previously pressed, and continues to be so: Pressed
- The button was previously pressed, but it has just become un-pressed: Released

You are proposing to make things even more confusing. What this code says is:

- The button WAS not pressed, but it just became pressed. Pressed.
- The button was previously pressed, and continues to be so: IsMouseDown
- The button was previously pressed, but it has just become un-pressed: Released

This may seem sublte and insignificant to you, but I don't think it is.

Reply:
Well they are only internal static variables to hold and control the flux of the mouse events. Say what names you want for them. They do what you need for the `ViewMouse.cs` handle them through the `IsMouseDown` property.